lib/r.lua

```diff
@@ -458,6 +485,50 @@ function util.split_ref(ref)
   return words[1], words[2]
 end
 
+function util.make_param(modulename, module, param, polyphony, controlspecoptions, name, paramformatter, paramparamset) -- added by @andrew
+  polyphony = polyphony or 1
+  paramparamset = paramparamset or params
+  paramcontrolspec = R.specs[module][param]
```

Comment:
`local` needed here to not globber the global namespace?

Reply:
In my mind since they were already parameters they were local - but adding `local` can't hurt!

Reply:
Sorry, I was a bit unclear, my comment was only about `paramcontrolspec` which I could not find in the argument list.

Lua is all right but the defaulting variables to globals... 👿